### PR TITLE
Refactor FXIOS-18221 [v124] Implement searchSettingsDidLoad action

### PR DESCRIPTION
**Pull request by imsanchit**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8195)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18221)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift [hunk 1]
```diff
@@ -116,6 +113,30 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
...
+        })
+    }
+
+    func unsubscribeFromRedux() {
+        store.unsubscribe(self)
+    }
+
+    func newState(state: SearchSettingsState) {
+        searchSettingsState = state
+        self.navigationItem.title = .Settings.Search.Title
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:134]: given that this isn't changing, should it be in a state?

cc @OrlaM

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift [hunk 2]
```diff
@@ -50,6 +52,7 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
 
     private var showDeletion = false
     private var sectionsToDisplay: [SearchSettingsTableViewController.Section] = []
+    private var searchSettingsState: SearchSettingsState
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:55]:
```suggestion
    private var searchSettingsState: SearchSettingsState?
```
This should be optional, and then assigned in `newState`

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift [hunk 3]
```diff
@@ -64,7 +67,7 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
     init(profile: Profile) {
         self.profile = profile
         model = profile.searchEngines
-
+        self.searchSettingsState = SearchSettingsState()
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:70]: We shouldn't init this here, it should be in `newState`

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift [hunk 4]
```diff
@@ -116,6 +113,30 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
...
+    }
+
+    func newState(state: SearchSettingsState) {
+        searchSettingsState = state
+        self.navigationItem.title = .Settings.Search.Title
+        
+        // To allow re-ordering the list of search engines at all times.
+        self.tableView.isEditing = true
+        // So that we push the default search engine controller on selection.
+        self.tableView.allowsSelectionDuringEditing = true
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:139]: Unless we're updating these here, we shouldn't be adding them ahead of time. Only when the state is actually doing something with them.

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift [hunk 5]
```diff
@@ -116,6 +113,30 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
     override func viewDidDisappear(_ animated: Bool) {
         super.viewDidDisappear(animated)
         setEditing(false, animated: false)
+        unsubscribeFromRedux()
+    }
+
+    // MARK: - Redux
+
+    func subscribeToRedux() {
+        store.dispatch(SearchSettingsAction.searchSettingsDidLoad)
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:122]: Should this be dispatching this search settings action, or should this be `store.dispatch(ActiveScreensStateAction.showScreen(.searchSettingsViewController))`? @OrlaM

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift (hunk 1 above)
Comment (OrlaM) [reply firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:134]: Only dynamic parts of the view should be updated here. Everything static can be set up somewhere else (init, view did load etc)

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift [hunk 6]
```diff
@@ -74,13 +77,7 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
...
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        navigationItem.title = .Settings.Search.Title
-
-        // To allow re-ordering the list of search engines at all times.
-        tableView.isEditing = true
-        // So that we push the default search engine controller on selection.
-        tableView.allowsSelectionDuringEditing = true
+        subscribeToRedux()
```
Comment (OrlaM) [inline firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:80]: This needs to be behind a feature flag. The old code should continue as it is today if the feature flag is disabled.

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift (hunk 5 above)
Comment (adudenamedruby) [reply firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:131]: Just double checking, on this, @OrlaM.

**Pull request closed without merging**